### PR TITLE
feat(php8 hanging comma) Add support for haning comma for functions, closures, arrow functions and methods

### LINKED ISSUE #1759: PHP8: Trailing comma in method signature
**Prettier 2.3.1**

**PHP Plugin 0.17.1**

**Input:**

```php
class T
{
    public function __construct(
        string $t1,
        string $t2,
        string $t3,
        string $t4,
        string $t5,
    ) {
    }
}
```

**Expected:**
Valid PHP 8 code; file should be formatted.

**Result:**

```
Parse Error : syntax error, unexpected ')', expecting T_VARIABLE on line 10
   8 |         string $t4,
   9 |         string $t5,
> 10 |     ) {
     |    ^
  11 |     }
  12 | }
  13 |
```

### PR DESCRIPTION
**Pull request by cseufert**
This patch requires the PHP parser support for hanging commas https://github.com/glayzzle/php-parser/pull/756

closes #1759

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   "dependencies": {
     "linguist-languages": "^7.5.1",
     "mem": "^8.0.0",
-    "php-parser": "https://github.com/glayzzle/php-parser#425c4dcc9f386596bdcd5b458503b3667a858141"
+    "php-parser": "https://github.com/moddengine/php-parser#c71ee4b5c7bab44dbaf5cb9e9bce7e69b45c23b8"
```
Comment (alexander-akait) [inline package.json:16]: Better to wait merge

**Review by alexander-akait — COMMENTED** [review]
Anyway good job!

File: package.json (hunk 1 above)
Comment (cseufert) [reply package.json]: @czosel I have updated this after you merged the changes into the parser

File: src/printer.js [hunk 2]
```diff
@@ -631,9 +631,12 @@ function printArgumentsList(path, options, print, argumentsKey = "arguments") {
   const node = path.getValue();
   const lastArg = getLast(args);
 
+  // 123abc
```
Comment (czosel) [inline src/printer.js:634]: Not needed i guess ;)

**Review by czosel — APPROVED** [review]
Looks great otherwise 👍

File: src/printer.js (hunk 2 above)
Comment (cseufert) [reply src/printer.js]: Oops, fixed.

**Comment by justim** [comment]
This change also adds a comma after "parameters" in a use block for a closure — which it should, but the resulting code can not be parsed again.

Input:

```php
<?php

$test = function (SomeClass $one, SomeOtherClass $two, AndAnotherOne $three)
  use ($one, $two, $three, $four, $five, $six, $seven, $eight, $nine, $ten, $eleven)
{
}
```

Output:

```php
<?php

$test = function (
    SomeClass $one,
    SomeOtherClass $two,
    AndAnotherOne $three,
) use (
    $one,
    $two,
    $three,
    $four,
    $five,
    $six,
    $seven,
    $eight,
    $nine,
    $ten,
    $eleven,
) {};
```

And using the output as input again give the following message:

```
Parse Error : syntax error, unexpected ')', expecting T_VARIABLE on line 19

  17 |     $ten,
  18 |     $eleven,
> 19 | ) {};
  20 |
```

See Prettier playgrounds: [before](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCDoA6BiBYwIbeLJUBhdAFtvbAAoAEv74zoKGIBwuAMZQAMowAJ5q1mDYEAC0AEyOINAAjoLw0P4uyn56cgogxCIQMKVs5WyVBgC+LFAU8DDoKGjwWJKgbNgwrSIdNZhgguN4IAA8APzGYAA6mJsAJDpjAAQAvPtSgpjRGJj7ABRxPlAesqPY+9tYhvt33lDINEiPz1eMAA7ugmPsAIKYChQ7TUJDITBQIHUaBQACUm32+2EyOubyR4N2oKJNDRRKk6EEEAp8AAbh9tkoAB5EkQMzBEqDwZTUGBEzDERk6TmvKBqDmYzDATZtEBtIA) and [after](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCDoA6BiBYwIbeLJUBhdAFtvbAAoAEv74zoKGIBwuAMZQAMowAJ5q1mDYEAC0AEyOINAAjoLw0P4uyn56cgogxCIQMKVs5WyVBgC+LFAU8DDoKGjwWJKgbNgwrSIdNZhgguN4IAA8APzGYAA6mJsAJDpjAAQAvPtSgpjRGJj7ABSb+-f7cT5QHrKj2PvbWIZ3D0-eUGQNCQr3enxgAHd0ExfvcAIKYCgI7TUJDITBQcHUaA-TAASn2wkxtyuDy+GJhpPuuyhlIeWJxdLJUnQgggTOpUngADdcfTtkoAB4cz4iXmYEXbKDwZTUGCSzDEPlknQS2GfKBqcWUgnANoAbk2IDaQA).

Is this a quick fix? Or do you want me to look into it?

File: package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
   "dependencies": {
     "linguist-languages": "^7.5.1",
     "mem": "^8.0.0",
-    "php-parser": "https://github.com/moddengine/php-parser#c71ee4b5c7bab44dbaf5cb9e9bce7e69b45c23b8"
+    "php-parser": "hhttps://github.com/glayzzle/php-parser#cca83041d938b972c7ef74648f29de03777b0ea4"
```
Comment (alexander-akait) [inline package.json]: @czosel here bug

Comment (justim) [reply package.json]: That's not the latest version you're looking at. It currently links to the right commit on `master`, and the latest version of this PR as well.

The parses doesn't understand the extra `,`.

**Comment by alexander-akait** [comment]
@justim your code should work, can you open an new issue (before please test it on the latest version)

**Pull request merged**